src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -184,12 +186,13 @@ bool c_typecheck_baset::gcc_types_compatible_p(
   }
   else
   {
-    if(type1==type2)
+    if(are_types_equal_ignoring_typedef(type1, type2))
     {
       // Need to distinguish e.g. long int from int or
       // long long int from long int.
       // The rules appear to match those of C++.
-
+      // Isn't this explictly handled by checking type1==type2 (since
```

Comment:
No it was meant to be a question, meant to pull it out when I created the PR to ask the relevant person.

@kroening - is there a reason for [this line](https://github.com/diffblue/cbmc/pull/358/files/30fb4621075d69a61ec6974acd76674ada4b8b0a..23aa4262cfa1b26d19b17befcc728e68cbcfa5f6#diff-0984c6664412e8900073e1dd107c6658R196), it seems to me that it is implied by [this line](https://github.com/diffblue/cbmc/pull/358/files/30fb4621075d69a61ec6974acd76674ada4b8b0a..23aa4262cfa1b26d19b17befcc728e68cbcfa5f6#diff-0984c6664412e8900073e1dd107c6658L187)